### ISSUE TITLE
Aria-hidden <label> elements 

### ISSUE THREAD

**Issue by shanderson** [dequelabs/axe-core#132]
NVDA and JAWS will associate a label to an input field even if there is an aria-hidden on the <label> element but this is identified as a missing programmatic label violation by axe. Is this really a violation?

**Comment by dylanb** [comment]
What mechanism is being used for the association? Can you provide an example?

**Comment by shanderson** [comment]
old fashioned for and id. I will try to find the example in the wild.

**Comment by shanderson** [comment]
<label aria-hidden="true" class="credit-card-field-label" id="ccExpDate" for="expirationMonth">Expiration Date</label>
                                    <span aria-hidden="true" class="required">&nbsp;*</span>
                                <select id="expirationMonth" name="expirationMonth" aria-required="true">
                                    
                                    <option value="" selected="selected">Select Month</option>

	<option value="1">1 - January</option>

	<option value="2">2 - February</option>

	<option value="3">3 - March</option>

	<option value="4">4 - April</option>

	<option value="5">5 - May</option>

	<option value="6">6 - June</option>

	<option value="7">7 - July</option>

	<option value="8">8 - August</option>

	<option value="9">9 - September</option>

	<option value="10">10 - October</option>

	<option value="11">11 - November</option>

	<option value="12">12 - December</option>


                                </select>

**Comment by dylanb** [comment]
@shanderson what screen readers read this out?

**Comment by shanderson** [comment]
NVDA and Jaws read the label. Just checked with VO and it didn't read.

**Comment by dsturley** [comment]
Based on the [spec](http://www.w3.org/TR/accname-aam-1.1/#mapping_additional_nd_te); the input should have an accessible name.  The relevant bit is 2A and [our implementation](https://github.com/dequelabs/axe-core/blob/master/lib/commons/text/accessible-text.js#L241)

> If the current node is hidden and is not referenced by aria-labelledby or aria-describedby, nor referenced by a native host language text alternative element or attribute, return the empty string.

In this instance, its referenced by a native host language text alternative attribute (`label for="id"`), therefore visibility should have no effect and thus continue.

**Comment by dylanb** [comment]
@dsturley did you see my discussion on Slack with Steve Faulkner and Alice on this? It seems as though there are contradictions between that spec and the two ARIA specs 1.0 and 1.1. Differences in the definition of hidden as well as the second half of the sentence you posted being (it seems) new.

**Comment by dylanb** [comment]
Here is the current support. I suggest we hold off until there is better support of the new algorithm.

https://lists.w3.org/Archives/Public/public-pfwg/2015Sep/0217.html
